### ISSUE TITLE
Optimize dist/vendor in prod

### ISSUE THREAD

**Issue by pierredewilde** [angular/angular-cli#1208]
1. OS?
```
Mac OS X El Capitan 10.11.5
```
2. ng --version
```
angular-cli: 1.0.0-beta.8
node: 6.2.2
os: darwin x64
````
3. Repro steps

`ng build -prod` creates bundle `main.js` containing minified angular 2 files. 

Vendor files are stored unminified in `dist/vendor` directory :

```
$ tree dist/vendor
├── es6-shim
│   └── es6-shim.js
├── reflect-metadata
│   ├── Reflect.js
│   ├── Reflect.ts
│   ├── reflect-metadata.d.ts
│   ├── test
│   │   ├── harness.ts
│   │   ├── reflect
│   │   │   ├── reflect-decorate.ts
│   │   │   ├── reflect-definemetadata.ts
│   │   │   ├── reflect-deletemetadata.ts
│   │   │   ├── reflect-getmetadata.ts
│   │   │   ├── reflect-getmetadatakeys.ts
│   │   │   ├── reflect-getownmetadata.ts
│   │   │   ├── reflect-getownmetadatakeys.ts
│   │   │   ├── reflect-hasmetadata.ts
│   │   │   ├── reflect-hasownmetadata.ts
│   │   │   └── reflect-metadata.ts
│   │   ├── run.ts
│   │   └── spec.ts
│   ├── typings
│   │   └── node.d.ts
│   └── typings.d.ts
├── systemjs
│   └── dist
│       └── system.src.js
└── zone.js
    └── dist
        └── zone.js
```

Instead of those multiple unminified files, another bundle (eg `vendors.js`) may be used with minified versions of :

```
vendor/es6-shim/es6-shim.js
vendor/reflect-metadata/Reflect.js
vendor/systemjs/dist/system.src.js
vendor/zone.js/dist/zone.js
```

This should simplify and optimise the -prod distribution.

Of course, `index.html` should be also simplified to reflect this change.

**Comment by kylecordes** [comment]
I think there is other work coming to replace es6-shim and reflect-metadata with core-js. That will reduce the file count by one... of course more is also needed eventually.  I wrote about some of this a while back in #987.

**Comment by filipesilva** [comment]
Closed as issue was made obsolete by #1455.
